docs/helm-chart/index.rst

```diff
@@ -140,3 +140,22 @@ will not start as the migrations will not be run:
 This is so these CI/CD services can perform updates without issues and preserve the immutability of Kubernetes Job manifests.
 
 This also applies if you install the chart using ``--wait`` in your ``helm install`` command.
+
+.. note::
+    While deploying this Helm chart with Argo, you might encounter issues with database migrations not running automatically on upgrade.
+
+To ensure database migrations with Argo CD, you will need to add:
+
+.. code-block::yaml
+
+    migrateDatabaseJob:
+        jobAnnotations:
+            "argocd.argoproj.io/hook": Sync
+
+and, equivalent configurations in case of Flux CD
+
+.. note::
+    This will ensure database migrations run when the Airflow Docker image is upgraded. This approach has a limitation in that the database migrations will run every time there is a ``Sync`` event in Argo. This is a trade-off for automation at the cost of some computational loss.
```

Comment:
```suggestion
This will ensure database migrations run when the Airflow Docker image is upgraded. This approach has a limitation in that the database migrations will run every time there is a ``Sync`` event in Argo. This is a trade-off for automation at the cost of some computational loss.
```

Probably doesn't need to be a note, as it's just a continuation of the stuff right above it.

Is this actually tied to image changes, or would any change kick it off (I'm not very familiar with Argo CD)?

Reply:
Other changes in the helm chart values for deployment will also kick it off, apart from the image changes. Which is why I added this a note as this does have some limitations.  I am happy to remove this as a note and convert it into normal text if you feel thats more appropriate